### ISSUE TITLE
Green Project (1274490)

### ISSUE THREAD

**Issue by ex5** [ValveSoftware/Proton#4026]
# Compatibility Report
- Name of the game with compatibility issues: Green Project
- Steam AppID of the game: 1274490

## System Information
- GPU: Intel Corporation HD Graphics 630
- Driver/LLVM version: Mesa 20.0.7
- Kernel version: 5.6.4-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/): [steam sysinfo](https://gist.github.com/anka-sirota/bfcfb7a3732260a6da23b63b58dafaae)
- Proton version: 5.0.9 (same issue with 5.11-GE-1-MF, 5.9-GE, 5.6-GE)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[Proton log, PROTON_LOG=1 %command%](https://gist.github.com/anka-sirota/7b538a457082d1969d04855897e9b405)

## Symptoms 

Crashes immediately on startup with the following error in the [Unity log](https://gist.github.com/anka-sirota/1b3ddbfb937055c298c71ed4db5f6345):
```
UnityPlayer.dll called a function with an invalid parameter in module UnityPlayer.dll at 0033:80b77b8e
```
Removing `kernel32.dll` changes the traceback, but the game still crashes.

The crash seems to be system config specific, this is not the first Unity game that fails like this but has green reports on ProtonDB.
It looks like it might have something to do with the following quirks of the config:

* two graphic cards/multiple sound devices, issue reproducible on both graphic cards;
* no pulseaudio, only ALSA.

**^ That said**, I'm not sure how to dig deeper and would appreciate some pointers for figuring out what exactly causes the crash and possibly finding a work around.

**Comment by FelixSchwarz** [comment]
I was able to get the game running by going back to proton 4.11-13. Not the best solution but at least the game runs again.

**Comment by kisak-valve** [comment]
Hello @FelixSchwarz, what specific Proton version did you test with? (Proton 4.0 isn't a valid version.) If you tell Steam to run the game with Proton 5.0-10 after running it with Proton 4.11-13, does it continue working?

**Comment by FelixSchwarz** [comment]
@kisak-valve No, unfortunately not. However I'm a bit confused as I got inconsistent results when trying different versions on that computer (Fedora 33 + Nvidia GT1030/binary driver) so I retried with this with an AMD system (F33 with kernel 5.11.7, RX570/open source drivers, flathub Steam).

- Proton Experimental / Proton 5.13-6: game stops running almost immediately, no output
- Proton 5.0-10: game seems to start normally (including hw GFX acceleration)

So even on the AMD system there seems to be a Proton regression. The problem is reproducible even after reboots. One puzzling issue is that the game graphics is extremely slow (as if it would use software rendering instead of hw acceleration) if I start the game again without a system reboot (maybe logging in would be enough but I did not test this). Therefore I made sure to always reboot the system before trying another Proton version.

I'd be happy to provide more logs or information. I can also compile stuff myself/bisect if there is some documentation how to do that with Steam/Proton.

I should note that the game somewhat works with my wine(-staging) 6.3 from Fedora (no winetricks, just running `wine "Green Project.exe"`). However many texts are not shown (e.g. "settings" dialog has no texts at all but the in-game texts are mostly present).

**Comment by kisak-valve** [comment]
It sounds like you encountered https://github.com/ValveSoftware/steam-runtime/issues/294 with your AMD system.
